doc/ui-guidelines.md

```diff
@@ -0,0 +1,254 @@
+
+BEFORE:
+```clojure
+(defn- view-internal
+  [_ _]
+  (let [pressed? (reagent/atom false)]
+    (fn
+      [{:keys [theme on-press on-long-press icon]}]
+      [rn/pressable
+       {:style               (style/main @pressed? theme)
+        :on-press            on-press
+        :on-press-in         #(reset! pressed? true)
+        :on-press-out        #(reset! pressed? nil)
+        :on-long-press       on-long-press}
+       [quo.icons/icon icon]])))
+
+(def view (theme/with-theme view-internal))
+```
+
+NOW:
+```clojure
+(defn view
+  [{:keys [on-press on-long-press icon]}]
+  (let [[pressed? set-pressed] (rn/use-state false)
+        theme                  (theme/use-theme-value)
+        on-press-in            (rn/use-callback #(set-pressed true))
+        on-press-out           (rn/use-callback #(set-pressed nil))]
+    [rn/pressable
+     {:style               (style/main pressed? theme)
+      :on-press            on-press
+      :on-press-in         on-press-in
+      :on-press-out        on-press-out
+      :on-long-press       on-long-press}
+     [quo.icons/icon icon]]))
+```
+
+
+- We no longer need to create an anonymous function for rendering. This removes unnecessary confusion and the need for specific knowledge on how it works and why it was needed.
+- `rn/use-state` is used instead of `reagent/atom`
+- State values no longer need to be dereferenced; they are accessible as regular symbols. This eliminates a common bug where the "@" symbol was inadvertently omitted.
+- `theme/with-theme` wrapper is not needed anymore, `(theme/use-theme-value)` hook can be used directly in the components
+- `:f>` not needed anymore, all components are functional by default
+- `rn/use-callback` hook should be used for anon callback functions
+
+> [!IMPORTANT]
+> DO NOT USE anon functions directly in the props
+ 
+BAD
+```clojure
+(defn view
+  []
+  (let [[pressed? set-pressed] (rn/use-state false)]
+    [rn/pressable
+     {:style               (style/main pressed?)
+      :on-press-in         #(set-pressed true)
+      :on-press-out        #(set-pressed nil)}]))
+```
+
+GOOD:
+```clojure
+(defn view
+  []
+  (let [[pressed? set-pressed] (rn/use-state false)
+        on-press-in            (rn/use-callback #(set-pressed true))
+        on-press-out           (rn/use-callback #(set-pressed nil))]
+    [rn/pressable
```

Comment:
hmm, is this usage of `use-callback correct?
Is there a difference between?
`(rn/use-callback #(set-pressed true))`
and
`(rn/use-callback #(set-pressed true) [])` ??

also it might be good to have an example where there is dependencies needed for the use-callback to be more effective.

Reply:
see: https://stackoverflow.com/questions/55026139/whats-the-difference-between-usecallback-with-an-empty-array-as-inputs-and-u

<img width="327" alt="Screenshot 2024-02-23 at 10 01 15" src="https://github.com/status-im/status-mobile/assets/22799766/c22f7f9c-88a9-4fe6-ada2-8b9cb5bf2dba">

Reply:
ah, I see you've handle it in the react-native file. But perhaps this is misleading? do we want to wrap the react hooks this way, might be confusing at times?

Reply:
yes it might be confusing with useEffect, idk, but in most cases callback should be just cached, so passing each time empty vector looks redundant, deps in useCallback are not optional, so we pass empy array in our use-callback wrapper

Reply:
for useEffect deps are optional, because there might be the cases when you want to run effect on each update, but for useCallback it's not the case, so idk, but for 80% of cases you will have to pass empty [] manually

Reply:
now i see your lates comments, yes for useEffect it was confusing for sure, so i reverted it, for useCallback i hope it will work fine

Reply:
Yeah I can see that. I guess it's how much do we want to abstract away from devs.
Imo it's probably better to go with as vanilla a react experience at the beginning so the team becomes familiar with the react api and then abstract away later on.
However not so strongly opinionated on this and either way is fine 👍